File: projects/data/r2dbc/src/main/java/com/gmail/unmacaque/spring/data/r2dbc/domain/Message.java
package com.gmail.unmacaque.spring.data.r2dbc.domain;

import org.springframework.data.annotation.Id;

public class Message {

	@Id
	private Long id;

	private String title;

	private String author;

	private String content;

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}

	public Long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public String getAuthor() {
		return author;
	}

	public void setAuthor(String author) {
		this.author = author;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}
}

File: projects/kotlin/src/main/kotlin/com/gmail/unmacaque/spring/kotlin/domain/Message.kt
package com.gmail.unmacaque.spring.kotlin.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
data class Message(
    val sender: String,
    val recipient: String,
    val content: String,
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) val id: Long = -1
)
